apps/src/applab/gridUtils.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * A couple of utility functions for dealing with our design mode grid.
+ */
+module.exports = {
+  GRID_SIZE: 5,
+  /**
+   * Given an element being dragged, determine the scaled x/y position of the
+   * top left corned, scaled to our visualization.
+   * @param {jQueryObject} draggingElement
+   * @return {object}
+   */
+  scaledDropPoint: function (draggingElement) {
+    var div = document.getElementById('designModeViz');
```

Comment:
A few thoughts:

1. Since we're only using this overlay in App Lab right now, the dependency on designModeViz in particular doesn't bother me.  It seems like a thing we can clean up if we ever try to use it for something else.

2. That said, I'm tempted to ask that we inject/cache this element reference (and make some other changes) for performance reasons.  You're now calling this function from inside `onSvgMouseMove_` and doing a `document.getElementById` call along with multiple `getBoundingClientRect` inside that method doesn't seem great.  The `this.mousePos_.matrixTransform(this.screenSpaceToAppSpaceTransform_);` done in the non-dragging case caches the appropriate transform for the current visualization scale, so we don't do all this stuff every frame.  It might even be accelerated in certain browsers.